rmw_fastrtps_cpp/include/rmw_fastrtps_cpp/TypeSupport_impl.h

```diff
@@ -514,6 +514,12 @@ void deserialize_array(
     (void)call_new;
     if (member->array_size_ && !member->is_upper_bound_) {
         deser.deserializeArray((T*)field, member->array_size_);
+    } else {
+        auto & data = *reinterpret_cast<typename GenericCArray<T>::type *>(field);
+        int32_t dsize;
```

Comment:
Looks like cppcheck is concerned about this not be initialized:

http://ci.ros2.org/job/ci_linux/1606/testReport/(root)/projectroot/cppcheck/

You could either give it a default value or use the no lint comment to squelch cppcheck.